x-pack/legacy/plugins/apm/public/new-platform/plugin.tsx

```diff
@@ -93,7 +92,6 @@ export class ApmPlugin
 
   // Take the DOM element as the constructor, so we can mount the app.
   public setup(_core: CoreSetup, plugins: ApmPluginSetupDeps) {
-    plugins.home.featureCatalogue.register(featureCatalogueEntry);
```

Comment:
This is in here for the NP plugin to register itself, and will run both as a shim and when we move it to a pure NP plugin. The legacy_register_feature.ts above is only run in the legacy plugin. 

We would only be removing this here if the featureCatalog.register is not wanted in the NP plugin. Is that the case?

Reply:
It is wanted, but currently it's registered twice because the `legacy_register_feature` file is pulled into the apm bundle as well. I traced back the change and this happens because the `home` ui export is a dependency of visualize which is loaded everywhere. It seems like this is not necessary and removing the import there also fixes this problem.